src/test/java/array/Array1dTest.java

```diff
@@ -0,0 +1,207 @@
+package array;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class Array1dTest {
+
+    @Test
+    void shouldCreateEmptyArrayByConstructor() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = new Array1d<>(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldCreateEmptyArrayByFactoryMethod() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = Array1d.of(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldVisitEmptyArray() {
+        // given
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() -> array.visit(i -> System.out.print(i + ", ")));
+    }
+
+    @Test
+    void shouldFillArrayWithGivenNumbers() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() ->
+                IntStream.range(start, end)
+                        .forEach(i -> array.insert(
+                                Coordinates1d.of(i),
+                                10 * i)));
+    }
+
+    @Test
+    @DisplayName("Should throw an exception at second insertion into already occupied cell")
+    void shouldThrowAnException() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates1d.of(i),
+                        10 * i));
+        // expect
+        assertThrows(IllegalArgumentException.class,
+                () -> array.insert(Coordinates1d.of(1), 100));
+    }
+
+    @Test
+    @DisplayName("Should throw an exception if index out of bound")
+    void shouldThrowAnExceptionAlso() {
+        // given
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertThrows(ArrayIndexOutOfBoundsException.class,
+                () -> array.insert(Coordinates1d.of(2 * size), 120));
+    }
+
+
+    @Test
+    void shouldVisitNonEmptyArray() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates1d.of(i),
+                        10 * i));
+        // expect
+        assertDoesNotThrow(() -> array.visit(i -> System.out.print(i + ", ")));
```

Comment:
maybe
1. visit every cell and concat all its value to a string
2. compare result with expected etalon
?

Reply:
`Visit` method returns `void`. How to implement this?

Reply:
but it take lambda as a param - you can pass a variable in lambda, like `StringBuilder` instance, populate it in lambda block, and than check the value after all cells are visited

Reply:
Nice! Will try

Reply:
Fixed 48758e